engine/gamesys/src/gamesys/resources/res_model.cpp

```diff
@@ -78,137 +63,73 @@ namespace dmGameSystem
-            return out + icount;
-        }
-    }
 
-    static uint16_t* WriteModelMeshIndices16(const dmRigDDF::Mesh* mesh, uint16_t start, uint16_t* out, uint32_t* num_written_indices)
-    {
-        if(mesh->m_IndicesFormat == dmRigDDF::INDEXBUFFER_FORMAT_32)
-        {
-            uint32_t icount = mesh->m_Indices.m_Count / 4;
-            uint32_t* idata = (uint32_t*)mesh->m_Indices.m_Data;
-            for(uint32_t i = 0; i < icount; ++i)
+            for (int c = 0; c < 4; ++c)
             {
-                out[i] = start + (uint16_t)idata[i];
+                out_write_ptr->color[c] = colors ? *colors++ : 1.0f;
             }
-            *num_written_indices = icount;
-            return out + icount;
-        }
-        else
-        {
-            uint32_t icount = mesh->m_Indices.m_Count / 2;
-            uint16_t* idata = (uint16_t*)mesh->m_Indices.m_Data;
-            for(uint32_t i = 0; i < icount; ++i)
+
+            for (int c = 0; c < 2; ++c)
             {
-                out[i] = start + idata[i];
+                out_write_ptr->uv0[c] = uv0 ? *uv0++ : 0.0f;
+                out_write_ptr->uv1[c] = uv1 ? *uv1++ : 0.0f;
             }
-            *num_written_indices = icount;
-            return out + icount;
+
+            out_write_ptr++;
         }
+
+        return out_write_ptr;
     }
 
-    static ModelResourceBuffers* CreateBuffers(dmGraphics::HContext context, const dmRigDDF::Mesh* ddf_mesh, bool support_32bit_indices, dmArray<dmRig::RigModelVertex>& scratch_buffer)
+    static ModelResourceBuffers* CreateBuffers(dmGraphics::HContext context, const dmRigDDF::Mesh* ddf_mesh, dmArray<dmRig::RigModelVertex>& scratch_buffer)
     {
         ModelResourceBuffers* buffers = new ModelResourceBuffers;
         memset(buffers, 0, sizeof(ModelResourceBuffers));
 
         uint32_t num_vertices = ddf_mesh->m_Positions.m_Count / 3;
-        if (num_vertices < ddf_mesh->m_Vertices.m_Count)
-            num_vertices = ddf_mesh->m_Vertices.m_Count;
 
+        bool support_32bit_indices = dmGraphics::IsIndexBufferFormatSupported(context, dmGraphics::INDEXBUFFER_FORMAT_32);
+
+        dmGraphics::Type index_element_type = dmGraphics::TYPE_UNSIGNED_SHORT;
+        void* index_buffer = ddf_mesh->m_Indices.m_Data;
         uint32_t num_indices = ddf_mesh->m_Indices.m_Count / 2;
         if (ddf_mesh->m_IndicesFormat == dmRigDDF::INDEXBUFFER_FORMAT_32)
+        {
             num_indices = ddf_mesh->m_Indices.m_Count / 4;
+            index_element_type = dmGraphics::TYPE_UNSIGNED_INT;
 
-        dmGraphics::Type index_element_type = dmGraphics::TYPE_UNSIGNED_SHORT;
-
-        void* index_buffer = 0;
+            // We've moved the runtime check to the build step instead
+            if (!support_32bit_indices)
+            {
+                dmLogError("The platform doesn't support 32 bit index buffers. See the setting 'model.split_large_meshes'");
+                return buffers;
+            }
+        }
```

Comment:
Output an error and return.
The loading doesn't fail, the model will be just not be there.